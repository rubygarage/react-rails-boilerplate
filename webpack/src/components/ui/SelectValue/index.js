import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/ui/Icon'

export function SelectValue({ children, value: { icon, label } }) {
  return (
    <div className="Select-value" title={label}>
      <div className="Select-value-label flex align-center">
        <Icon type={icon} className="font-20 mr-10" />
        <span className="select-value-text">{children}</span>
      </div>
    </div>
  )
}

SelectValue.propTypes = {
  children: PropTypes.node,
  value: PropTypes.shape({
    icon: PropTypes.string,
    label: PropTypes.string
  }).isRequired
}

export default SelectValue
