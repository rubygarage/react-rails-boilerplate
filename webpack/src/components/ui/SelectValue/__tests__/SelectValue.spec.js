import React from 'react'
import { shallow } from 'enzyme'
import SelectValue from 'components/ui/SelectValue'

describe('<SelectValue />', () => {
  const defaultProps = {
    value: { label: 'Audio', icon: 'icon-audio-circle' }
  }

  it('renders select value with icon', () => {
    const component = shallow(
      <SelectValue {...defaultProps} />
    )

    expect(component).toMatchSnapshot()
  })
})
