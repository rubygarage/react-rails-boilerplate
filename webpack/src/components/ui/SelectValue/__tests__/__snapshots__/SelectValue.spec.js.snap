// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectValue /> renders select value with icon 1`] = `
<div
  className="Select-value"
  title="Audio"
>
  <div
    className="Select-value-label flex align-center"
  >
    <Icon
      className="font-20 mr-10"
      type="icon-audio-circle"
    />
    <span
      className="select-value-text"
    />
  </div>
</div>
`;
